src/Host/Monitor/Impl/BrokerManager.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Common.Core;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.R.Host.Monitor {
+    public class BrokerManager {
+        private const string RHostBroker = "Microsoft.R.Host.Broker";
+        private static string RHostBrokerExe = $"{RHostBroker}.exe";
+        private static string RHostBrokerConfig = $"{RHostBroker}.Config.json";
+        private static Process _brokerProcess;
+
+        public static bool AutoRestart { get; set; }
+        public static int AutoRestartMaxCount {
+            get {
+                return Properties.Settings.Default.AutoRestartMaxCount;
+            }
+            set {
+                Properties.Settings.Default.AutoRestartMaxCount = value;
+            }
+        }
+
+        private static int _autoRestartCount = 0;
+
+        static BrokerManager() {
+            AutoRestart = true;
+        }
+
+        public static Task<int> CreateOrAttachToBrokerInstanceAsync(ILogger logger = null) {
```

Comment:
Do we want to protect from multiple calls? I.e. if task is already created, don't create another one.